server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -71,6 +74,10 @@ void setMessageListener(TransportMessageListener listener) {
         }
     }
 
+    void setSlowLogThreshold(TimeValue slowLogThreshold) {
```

Comment:
Admittedly the chain of setters here is a little hacky but:
1. Doing it this way made this a much much smaller change than passing the cluster settings directly to the `InboundHandler`
2. I'd do a follow-up of this logic for the outbound path and that would require handling the threshold setting in the `TransportService` anyway.